### DISCUSSION TITLE
Option to fill in top AutoComplete match when hitting Enter / Autofill

### DISCUSSION THREAD

**Discussion by iron3oxide** [lowdefy/lowdefy#1265] [Ideas and feature requests]
As mentioned previously, I'd like to convert a PySide6/Qt app into a web app. Autocomplete in Qt has the nice feature that hitting Enter inside such a widget automatically fills in the top match the autocomplete function found for the chars already in the input. 

I'd like to achieve something similar in lowdefy, however the option to enable this doesn't seem to exist yet. Since the lowdefy AutoComplete widget does have a logical order of matches (as seen in the dropdown when typing), this shouldn't be hard to achieve (unless I'm missing some kind of hidden complexity here).

**Comment by Gervwyk** [comment]
Hi @iron3oxide 
So currently with AutoComplete, you can type your option, hit down then enter and it will complete the list with first option. Which I guess is very close to what you are looking to achieve or not? 

However to set the value when enter is hit, and only when the page focus is on the AutoComplete input field will require additional code and events to implemented.. I have some ideas on how to add this functionality to allow for keyboards shortcuts to be specified, however, it is still very much unspecified. 

If you are confident with js, you should be able to write v4 action plugin to do this for the specific use case, which does not have to be very complex. Let me if you'd consider this, then we can discuss further.

**Reply by iron3oxide** [reply]
What I'm looking for is that when I type in "Ger" and the top match is e.g. "Germany", I only have to hit Enter and "Germany" replaces "Ger" in the input field. This does currently not work for me, neither in the docs nor in code. 

Sadly I'm not very confident with js (yet), lowdefy is kind of an excuse for me not to change that tbh :sweat_smile:. Might have to do so in the future though...

**Reply by Gervwyk** [reply]
Haha. Makes sense. So instead of simply pressing enter, try pressing down arrow then enter.. Lil more complex UX, but does the same thing.

**Reply by iron3oxide** [reply]
That's an okay-ish solution I guess, it's not as straightforward as I'd wish it to be, but if my ideal solution is way more complex to achieve, it'll have to do. 

A little explanation: I will not be the sole or even main user of the app, my colleagues (mostly non-technical people) will be. I have found many of them to use inefficient/slow ways of interacting with the app (e.g. selecting input fields in a form with the mouse instead of tabbing), wherever it wasn't abundantly obvious what the "fast" way to achieve something is. I did not observe this in the Qt Autocomplete fields however, as they were sort of "user-proof" and one quickly understood how to work with them efficiently. 

Knowing said users, they will either type in the whole text each time or select the top match by mouse click unless instructed to do otherwise (and even then it's very much possible some will continue doing the former). That's kind of my gripe here: the lowdefy AutoComplete widget could be more "obvious" in how to work with it, even I had to spend some time wondering what to do to make it work as more than just a suggestion widget.

**Reply by iron3oxide** [reply]
My apologies, #1270 just made me realize that the Selector widget actually does what I wanted the Autocomplete widget to do (fill out the top match when hitting Enter). I should have specified that I intended for only a set of options to be allowed, I can now see that this is where you distinguish it from the Autocomplete widget and (since other options are allowed in the latter) why there is no automatic fill-in there on hitting Enter. 

Maybe this is not the best place to document that design choice though, what about adding something like "... and want to be able to select the top match by hitting Enter,..." to the note in the Autocomplete doc page that references the Selector block?

**Reply by Gervwyk** [reply]
Sweet. I agree that would be a nice addition, would you mind adding it to this file https://github.com/lowdefy/lowdefy/blob/main/packages/docs/blocks/input/AutoComplete.yaml 
Then making a PR with the change for me to review? 
All community contributions are amazing 🙏

**Reply by iron3oxide** [reply]
done, see #1283
